plutus-core/testlib/UntypedPlutusCore/Test/Term/Construction.hs

```diff
@@ -0,0 +1,515 @@
+lessThanEqualsInteger i j = builtin LessThanEqualsInteger `app` i `app` j
+
+appendByteString :: T -> T -> T
+appendByteString b1 b2 = builtin AppendByteString `app` b1 `app` b2
+
+consByteString :: T -> T -> T
+consByteString i b = builtin ConsByteString `app` i `app` b
+
+sliceByteString :: T -> T -> T -> T
+sliceByteString start len bs =
+  builtin SliceByteString `app` start `app` len `app` bs
+
+lengthOfByteString :: T -> T
+lengthOfByteString bs = builtin LengthOfByteString `app` bs
+
+indexByteString :: T -> T -> T
+indexByteString bs i = builtin IndexByteString `app` bs `app` i
+
+equalsByteString :: T -> T -> T
+equalsByteString b1 b2 = builtin EqualsByteString `app` b1 `app` b2
+
+lessThanByteString :: T -> T -> T
+lessThanByteString b1 b2 = builtin LessThanByteString `app` b1 `app` b2
+
+lessThanEqualsByteString :: T -> T -> T
+lessThanEqualsByteString b1 b2 =
+  builtin LessThanEqualsByteString `app` b1 `app` b2
+
+sha2_256 :: T -> T
+sha2_256 bs = builtin Sha2_256 `app` bs
+
+sha3_256 :: T -> T
+sha3_256 bs = builtin Sha3_256 `app` bs
+
+blake2b_256 :: T -> T
+blake2b_256 bs = builtin Blake2b_256 `app` bs
+
+verifyEd25519Signature :: T -> T -> T -> T
+verifyEd25519Signature pk msg sig =
+  builtin VerifyEd25519Signature `app` pk `app` msg `app` sig
+
+verifyEcdsaSecp256k1Signature :: T -> T -> T -> T
+verifyEcdsaSecp256k1Signature pk msg sig =
+  builtin VerifyEcdsaSecp256k1Signature `app` pk `app` msg `app` sig
+
+verifySchnorrSecp256k1Signature :: T -> T -> T -> T
+verifySchnorrSecp256k1Signature pk msg sig =
+  builtin VerifySchnorrSecp256k1Signature `app` pk `app` msg `app` sig
+
+appendString :: T -> T -> T
+appendString s1 s2 = builtin AppendString `app` s1 `app` s2
+
+equalsString :: T -> T -> T
+equalsString s1 s2 = builtin EqualsString `app` s1 `app` s2
+
+encodeUtf8 :: T -> T
+encodeUtf8 s = builtin EncodeUtf8 `app` s
+
+decodeUtf8 :: T -> T
+decodeUtf8 bs = builtin DecodeUtf8 `app` bs
+
+ifThenElse :: T -> T -> T -> T
+ifThenElse cond t f = force (builtin IfThenElse) `app` cond `app` t `app` f
+
+chooseUnit :: T -> T -> T
+chooseUnit unitVal x = force (builtin ChooseUnit) `app` unitVal `app` x
+
+trace :: T -> T -> T
+trace msg x = force (builtin Trace) `app` msg `app` x
+
+fstPair :: T -> T
+fstPair p = force (force (builtin FstPair)) `app` p
+
+sndPair :: T -> T
+sndPair p = force (force (builtin SndPair)) `app` p
+
+chooseList :: T -> T -> T -> T
+chooseList xs t f = force (builtin ChooseList) `app` xs `app` t `app` f
+
+mkCons :: T -> T -> T
+mkCons x xs = force (builtin MkCons) `app` x `app` xs
+
+headList :: T -> T
+headList xs = force (builtin HeadList) `app` xs
+
+tailList :: T -> T
+tailList xs = force (builtin TailList) `app` xs
+
+nullList :: T -> T
+nullList xs = force (builtin NullList) `app` xs
+
+chooseData :: T -> T -> T -> T -> T -> T -> T
+chooseData d constr m xs i b =
+  force (builtin ChooseData)
+    `app` d
+    `app` constr
+    `app` m
+    `app` xs
+    `app` i
+    `app` b
```

Comment:
Also: the heavy use of infix functions like `` `app` `` looks visually unappealing.

Reply:
I guess its a matter of taste 🤷🏼‍♂️ 
I was thinking of having a variation of `$` operator for infix applications, something like:
```haskell
  term
    $. m
    $. xs
    $. i
    $. b
```